cvapipe_analysis/steps/shapemode/shapemode.py

```diff
@@ -24,206 +20,219 @@
-        # The feature calculation will be incorporated as an extra step in this
-        # workflow
-        path_to_manifest = f"{self.step_local_staging_dir}/allcells_shapespace1.csv"
-        if not os.path.exists(path_to_manifest):
-
-            _ = quilt3.Package.browse(
-                name="matheus/cell_shape_variation_shapespace1",
-                registry="s3://allencell-internal-quilt",
-            ).fetch(self.step_local_staging_dir)
-
-        """
-        --------------
-        PRE-PROCESSING
-        --------------
-        """
-
-        # Read the dataframe
-        df = pd.read_csv(path_to_manifest, index_col=0, nrows=(1024 if debug else None))
-        print(f"Shape of raw data: {df.shape}")
-
-        # Save tava table with number of mitotic vs. intephase cells
-        df["is_mitotic"] = df.cell_stage != "M0"
-        dataset_summary_table(
-            df=df,
-            levels=["is_mitotic"],
-            factor="structure_name",
-            rank_factor_by="meta_fov_image_date",
-            save=self.step_local_staging_dir / "allcells",
-        )
-        # Exclude mitotics
-        df = df.loc[df.is_mitotic == False]
-        df = df.drop(columns=["is_mitotic"])
-
-        print(f"Shape of data without mitotis: {df.shape}")
+    def run(
+        self,
+        debug: bool = False,
+        overwrite: bool = False,
+        remove_mitotics: bool = True,
+        no_structural_outliers: bool = False,
+        **kwargs
+    ):
+        
+        # Load feature dataframe
+        path_to_metadata_manifest = self.project_local_staging_dir / 'loaddata/manifest.csv'
+        df_meta = pd.read_csv(path_to_metadata_manifest, index_col='CellId')
+        # Drop unwanted columns
+        df_meta = df_meta[            
+            [c for c in df_meta.columns if not any(s in c for s in ['mem_','dna_','str_'])]
+        ]
+        log.info(f"Shape of metadata: {df_meta.shape}")
+        
+        # Load feature dataframe
+        path_to_features_manifest = self.project_local_staging_dir / 'computefeatures/manifest.csv'
+        df_features = pd.read_csv(path_to_features_manifest, index_col='CellId')
+        log.info(f"Shape of features data: {df_features.shape}")
+        
+        # Make necessary folders
+        tables_dir = self.step_local_staging_dir / 'tables'
+        tables_dir.mkdir(exist_ok=True)
 
         # Perform outlier detection based on Gaussian kernel estimation
-        dir_output_outliers = self.step_local_staging_dir / "outliers"
-        dir_output_outliers.mkdir(parents=True, exist_ok=True)
-        if True:
-            df_outliers = outliers_removal(df=df, output_dir=dir_output_outliers)
-            df_outliers.to_csv(self.step_local_staging_dir / "manifest_outliers.csv")
+        outliers_dir = self.step_local_staging_dir / 'outliers'
+        outliers_dir.mkdir(exist_ok=True)
+
+        # Mitotic removal
+        
+        if 'cell_stage' in df_meta.columns:
```

Comment:
should there be an `else` statement here?

Reply:
Not really. If there is no such column, all cells are used for downstream analysis.